libs/SmartSync/SmartSync/Classes/Manager/SFSmartSyncSyncManager.m

```diff
@@ -418,6 +418,9 @@ - (SFSyncState*)syncUpWithTarget:(SFSyncUpTarget *)target
                          options:(SFSyncOptions *)options
                         soupName:(NSString *)soupName
                      updateBlock:(SFSyncSyncManagerUpdateBlock)updateBlock {
+    if (target == nil) {
```

Comment:
Who is calling this with nil? How was it working before?
I think something is failing before the call to this method (maybe a regression introduced by my recent changes https://github.com/forcedotcom/SalesforceMobileSDK-iOS/pull/1892). I think this change is just masking the issue.